packages/insomnia/src/main/network/libcurl-promise.ts

```diff
@@ -332,7 +332,9 @@ export const createConfiguredCurlInstance = ({
   }
   const { validateSSL } = settings;
   if (!validateSSL) {
-    curl.setOpt(Curl.option.SSL_VERIFYHOST, 0);
+    if (process.platform !== 'darwin') {
```

Comment:
```suggestion
    // node-libcurl 2.4.1-4 macOS is using `--with-secure-transport` TLS backend to be able to read the keychain which interrupts VERIFY_HOST
    if (process.platform !== 'darwin') {
```
<img width="827" alt="image" src="https://github.com/Kong/insomnia/assets/3679927/8f269b71-e39c-44b8-b5ff-ae17d7cefd75">